### PR TITLE
updates developersetup and adding general notes on the workflows

### PR DESCRIPTION
**Pull request by xmonader**
### Description

updates developersetup and adding general notes on the workflows, also extending the makefile with checks and tests targets and adding more files to the wiki directory

### Changes

updates developersetup and adding general notes on the workflows, also extending the makefile with checks and tests targets

### Related Issues

- #454 

### Checklist

- [ ] Tests included
- [ ] Build pass
- [x] Documentation
- [ ] Code format and docstrings

### CODE REVIEW

File: wiki/workflows.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+## test workflow
+Runs against terraform matrix (only version 1.0.8) and only runs the tests in internal/provider directory
+
+
+TODO: check terraform versions
+
+## terratests workflow
+Runs the integration tests, it requires yggdrasil installation
+
```
Comment (AlaaElattar) [inline wiki/workflows.md:13]: It runs also the mock tests

File: wiki/developersetup.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+## Developer setup
+
+- should be using vscode
+- make sure to have `git`, `make`, `taskfile` installed
+- `make submodules` to populate the git submodules in the project
+
+
+## Before committing code
+- Make sure to run the using `make checks`
```
Comment (rawdaGastan) [inline wiki/developersetup.md:9]: run the ..?

File: wiki/developersetup.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- Make sure tests pass using `make unittests`
+
+## Integration tests
+Integration tets happen on the repository on the PRs
```
Comment (rawdaGastan) [inline wiki/developersetup.md:13]: tets -> tests

**Review by rawdaGastan — CHANGES_REQUESTED** [review]

File: wiki/workflows.md (hunk 1 above)
Comment (xmonader) [reply wiki/workflows.md:13]: I double checked, it runs with the `make unittests` target, I updated the docs

**Review by rawdaGastan — APPROVED** [review]

**Review by AbdelrahmanElawady — APPROVED** [review]

**Pull request merged**